backend/src/user_commands.py

```diff
@@ -20,45 +29,100 @@ def printCmd(cmdDict):
 
 def CMD_Add(cmdDict):
     printCmd(cmdDict)
+    # We Add funds to account, timestamp this action then log UserCommand
+    cmdDict['timestamp'] = str(int(time.time()))
+    log.logEvents['userCommand'](cmdDict)
+    # The users account is modified, log this action
+    log.logEvents['accountTransaction'](cmdDict)
 
 def CMD_Quote(cmdDict):
     printCmd(cmdDict)
+    # query the legacy quoteserver -> should return price of the stock
+    mockPrice = "10.00"
+    mockCryptoKey = "IRrR7UeTO35kSWUgG0QJKmB35sL27FKM7AVhP5qpjCgmWQeXFJs35g"
+    cmdDict['price'] = mockPrice
+    cmdDict['cryptokey'] = mockCryptoKey
+    cmdDict['timestamp'] = str(int(time.time()))
+    log.logEvents['quoteServer'](cmdDict)
 
 def CMD_Buy(cmdDict):
     printCmd(cmdDict)
+    buyAmtQ.put(cmdDict['amount'])
+    cmdDict['timestamp'] = str(int(time.time()))
+    log.logEvents['userCommand'](cmdDict)
 
 def CMD_CommitBuy(cmdDict):
     printCmd(cmdDict)
+    cmdDict['timestamp'] = str(int(time.time()))
+    log.logEvents['userCommand'](cmdDict)
+    # The users account is modified after committing to buying the stock, log this action
+    cmdDict['timestamp'] = str(int(time.time()))
```

Comment:
duplicate line on 56.

Reply:
There are 2 dfferent timestamps for the different events (first userCommand when the user issues the command then accountTransaction when the buy is committed). I wasn't sure how long the logic between them would take so I figured timestamp it twice?

Reply:
I see! yep fair enough!